exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/StringFunctions.java

```diff
@@ -268,7 +284,13 @@ public void eval() {
 
     @Override
     public void setup() {
-      matcher = java.util.regex.Pattern.compile(org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(pattern.start,  pattern.end,  pattern.buffer)).matcher("");
+      CharSequenceWrapper patternWrapper = new CharSequenceWrapper(pattern.start, pattern.end, pattern.buffer);
+      char[] chars = new char[patternWrapper.length()];
+      for(int i=0; i<patternWrapper.length(); i++){
+          chars[i] = patternWrapper.charAt(i);
+      }
+      String patternChars = new String(chars);
```

Comment:
I can not see why we want to use CharSequenceWrapper here. You created String object, same as prior method. Any benefit of using CharSequenceWrapper?